include/bluetooth/scan.rst

```diff
@@ -98,6 +98,17 @@ Check the following table for the details on the behavior of the available filte
 |              | If not all of these types match, the ``not found`` callback is triggered.                                 |
 +--------------+-----------------------------------------------------------------------------------------------------------+
 
+Connection attempts filter
+==========================
+
+After a scanning is started by the application some peripheral devices can disconnect immediately for some reasons during ongoing scanning session.
+This may result in a loop of the connection and disconnection events.
+To avoid this loop you can add the device the Connection attempts filter which limits number of the connection tries.
+Use the :option:`CONFIG_BT_SCAN_CONNECTION_ATTEMPTS_FILTER` to enable this filter.
```

Comment:
I think it would be good to mention that we count all the disconnections (not just the ones that occur quickly after the connection is established).
The application is responsible for cleaning the blacklist. If the applications does not clean the counters, the device will be unable to connect with a peripheral after a few disconnections related to successful connections and that would be strange from the user perspective.

Reply:
The scanning session could be a bit misleading since the module does not clean the blacklist automatically (its better to let application do it).

Reply:
I added some more description and split functionality between the blacklist feature and the connection attempts filter.